### DISCUSSION TITLE
How to propose changes to the site

### DISCUSSION THREAD

**Discussion by RobertAKARobin** [Geneticure/geneticure.com#34] [General]
## The advantages of using GitHub

- We have a record of every single change, including proposed changes
- We can easily revert any changes that are problematic
- We can comment on specific lines, instead of having to write "On line X of page Y where it says Z, we should change it to W."
- Everyone can see everyone else's changes/proposals
- We can respond to other people's suggestions
- No unwieldy email threads

## What to do

- Go to the desired repo
    - A 'Git repository' or "repo" is all of the files/folders that compose a codebase.  For example, the repo for the website is https://github.com/Geneticure/geneticure.com, and it contains all the HTML/text/images that make up the website.
    - You can see all our repos at https://github.com/orgs/Geneticure/repositories
- Navigate to the relevant page/text
    - For the website
        - The first place to look is in [`src/pages`](https://github.com/Geneticure/geneticure.com/tree/main/src/pages). Each `.astro` file in there corresponds to one of the "routes" on the website, e.g. /team, /providers, /privacy. The main page is `index.astro`. If the text you're looking for isn't in there, it may have been put in a separate file, in which case...
        - Otherwise, look in [`src/text`](https://github.com/Geneticure/geneticure.com/tree/main/src/text). This contains a lot of long-form text and things that are in lists, like FAQs, team bios, media links, etc.
- Open the relevant file by clicking on it
    - You'll probably see a lot of code in there, but it should be very easy to find the written copy. Ignore the code
- Make your suggestion:

### Propose a general change as an "Issue"

When looking at a file, click a line number on the left. (If you don't see line numbers, click the little '< >' button to display line numbers.) Then, click the little '`...`' button that shows to the left of the line. It expands into a menu. Click 'Reference in new issue'
![unnamed-4](https://user-images.githubusercontent.com/6268177/193071941-ae14582c-6bd6-49b3-b2af-d3f13f08e9d2.jpg)

#### Part 1: Create a new "branch"

A "branch" is like a copy of the main codebase (the "trunk") with some changes added to it. 

- Click the little pencil icon for 'Edit this file'
![unnamed](https://user-images.githubusercontent.com/6268177/193071944-bb2f4681-d96a-4eb4-a244-2db18542a3d0.jpg)
- Make changes / add comments right in the code.
    - Helpful tip: If the code goes off the side of the page, turn on 'soft line wrap'
    ![Screen Shot 2022-09-29 at 10 22 33 AM](https://user-images.githubusercontent.com/6268177/193072851-2ec2aeca-7c19-4d83-90f7-60ba88418e85.jpg)
- Scroll to the bottom and click 'Propose changes'
![unnamed-1](https://user-images.githubusercontent.com/6268177/193071930-0224e964-8594-4db7-b81b-584ded264bb3.jpg)
- You've created a branch. You can now preview your changes on a dev site:
    - Copy your branch name:
    ![Screen Shot 2022-09-30 at 11 25 50 AM](https://user-images.githubusercontent.com/6268177/193315784-f39f8d79-83f6-4962-b9f5-b215aab40d87.jpg)
    - In a new tab, paste this URL: `https://BRANCH--geneticure.netlify.app`
    - Replace the `BRANCH` with the branch name you copied.
        - In the example above, the branch name is `RobertAKARobin-patch-1`, so the URL is `https://RobertAKARobin-patch-1--geneticure.netlify.app`
    - It may take a few seconds for Netlify to build the page. Just keep refreshing until it shows up.

##### Part 1.1. Adding more changes to your branch

If you want, you can add more changes to your branch, and then create the PR when you're good and ready.

- When looking at a file/folder, switch to the branch you created by clicking the branch menu and then clicking on the name of your branch:
    ![Screen Shot 2022-09-30 at 11 33 25 AM](https://user-images.githubusercontent.com/6268177/193316451-5271e4ef-af47-4200-9ec5-fe12b1a476e9.jpg)
- When you finish editing this file, you should see that above the 'Commit changes' button it has selected 'Commit directly to the [branch name] branch.'
    ![Screen Shot 2022-09-30 at 11 34 53 AM](https://user-images.githubusercontent.com/6268177/193316706-3f899cd7-f8ee-40b1-bf5b-61caaeda7f37.jpg)

#### Part 2: Create a PR

Only create a PR when you're ready for Robin to review your changes and add them to the live site.

- Go to the 'Pull requests' tab and click 'New pull request'. (If you see a yellow box saying '[Your branch] had recent pushes...' then you can just click the 'Compare & pull request' button.)
    ![Screen Shot 2022-09-30 at 11 41 05 AM](https://user-images.githubusercontent.com/6268177/193317552-7781dde9-c113-4ac0-beea-4a04a36bfc00.jpg)
- From the 'compare:' menu, select your branch, then click 'Create pull request'
    ![Screen Shot 2022-09-30 at 11 42 37 AM](https://user-images.githubusercontent.com/6268177/193318030-916f506b-995b-4414-a88f-1037a9f1a590.jpg)
- Click 'Create pull request'. (Note that this is the page you see after creating a branch, so if you want you can skip the above steps.)
![unnamed-2](https://user-images.githubusercontent.com/6268177/193071934-69b5769f-eb7d-4199-b0fa-0eadb5b2938c.jpg)
- Done. Whenever someone creates a new PR we should all get notified. You can comment on a PR in the 'Conversation' tab, or on a specific line within the 'Files changed' tab by hovering over a line and clicking the little blue plus sign.
![unnamed-3](https://user-images.githubusercontent.com/6268177/193071938-cbf5d16e-9e8c-4776-8653-3ed77990ee3e.jpg)

**Comment by RobertAKARobin** [comment]
@Carl362 @snydreric @scottsnyder01 Forgot to mention that when you submit a PR, Netlify automatically creates a preview site for you on which you can see your changes. I just updated these instructions to include how to access it.

**Comment by scottsnyder01** [comment]
Oh that’s slick. Thanks

Scott Snyder
952.210.5851
________________________________

**Comment by RobertAKARobin** [comment]
...and you can actually view the preview site without submitting a PR, _and_ you can make changes to multiple files before submitting a PR. Just updated the instructions again.
